exist-core/src/main/java/org/exist/xquery/value/DoubleValue.java

```diff
@@ -327,15 +294,11 @@ public ComputableValue plus(final ComputableValue other) throws XPathException {
 
     @Override
     public ComputableValue mult(final ComputableValue other) throws XPathException {
-        switch (other.getType()) {
-            case Type.DOUBLE:
-                return new DoubleValue(getExpression(), value * ((DoubleValue) other).value);
-            case Type.DAY_TIME_DURATION:
-            case Type.YEAR_MONTH_DURATION:
-                return other.mult(this);
-            default:
-                return mult((ComputableValue) other.convertTo(getType()));
-        }
+        return switch (other.getType()) {
```

Comment:
Same comments as the above `switch` statement

Reply:
I do like the new syntax, here much more.

Reply:
@adamretter I would prefer to use of the new switch as function approach instead of the old style here too..